src/plugins/discover/public/plugin.tsx

```diff
@@ -419,32 +406,32 @@ export class DiscoverPlugin
     // initializeServices are assigned at start and used
     // when the application/embeddable is mounted
 
-    const { uiActions } = plugins;
-
-    const services = buildServices(
-      core,
-      plugins,
-      this.initializerContext,
-      this.locator!,
-      this.contextLocator!,
-      this.singleDocLocator!
-    );
-
-    uiActions.registerTrigger(SEARCH_EMBEDDABLE_CELL_ACTIONS_TRIGGER);
-
     const viewSavedSearchAction = new ViewSavedSearchAction(core.application);
-    uiActions.addTriggerAction('CONTEXT_MENU_TRIGGER', viewSavedSearchAction);
-    setUiActions(plugins.uiActions);
 
+    plugins.uiActions.addTriggerAction('CONTEXT_MENU_TRIGGER', viewSavedSearchAction);
+    plugins.uiActions.registerTrigger(SEARCH_EMBEDDABLE_CELL_ACTIONS_TRIGGER);
+    setUiActions(plugins.uiActions);
     injectTruncateStyles(core.uiSettings.get(TRUNCATE_MAX_HEIGHT));
 
+    const isDev = this.initializerContext.env.mode.dev;
+
+    const getDiscoverServicesInternal = async () => {
+      return this.getDiscoverServices(core, plugins);
+    };
+
     return {
       locator: this.locator,
       customize: createCustomizeFunction(this.profileRegistry),
-      useDiscoverMainRoute: (props?: UseDiscoverMainRouteProps) => {
-        return useDiscoverMainRouteInternal({
-          services: { ...services, ...props?.services },
-        });
+      DiscoverContainer: (props: DiscoverContainerProps) => {
+        const { overrideServices, ...restProps } = props;
```

Comment:
I am okay either ways. I generally prefer the explicit spreading of props but I will change it as you proposed.

Reply:
Done [here](https://github.com/elastic/kibana/pull/160036/files#diff-f4a8635de3049ee76e93df8b13eff18f08a042260eccda287d542ef247aecd6bR425).

Reply:
Apologies, I didn't really explain what I meant by this. I was wondering if it was necessary to explicitly extract `overrideServices` or if we could just spread the whole props param like this since it includes `overrideServices` anyway:
```ts
DiscoverContainer: (props: DiscoverContainerProps) => {
  return (
    <DiscoverContainerInternal
      getDiscoverServices={getDiscoverServicesInternal}
      isDev={isDev}
      {...props}
    />
  );
},
```

With that said, this is a super minor code style suggestion and not something I'd want to block this PR over, so I'd say it's fine the way it is 🙂. I'm hoping to look into if we can do anything on the Discover side about how we had to handle these services in `DiscoverContainer` anyway, and this is the kind of minor cleanup I could slip into that PR if I remember: https://github.com/elastic/kibana/blob/38c6d1ae59aa6d8dcb60d55e2d127a0eb8e9975d/src/plugins/discover/public/components/discover_container/discover_container.tsx#L60-L64